### PR TITLE
fix(core): when a gesture detected, a followed by click shouldn't trig

### PR DESCRIPTION
**Pull request by somebodyLi**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `LV_ITERATE_ROOTS` list in [`lv_gc.h`](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_gc.h) and mark the variable with `GC_ROOT(variable)` when it's used. See a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the followings needs to be followed (see a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)): 
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

File: src/core/lv_indev.c [hunk 1]
```diff
@@ -971,8 +971,8 @@ static void indev_proc_release(_lv_indev_proc_t * proc)
         lv_event_send(indev_obj_act, LV_EVENT_RELEASED, indev_act);
         if(indev_reset_check(proc)) return;
 
-        /*Send CLICK if no scrolling*/
-        if(scroll_obj == NULL) {
+        /*Send CLICK if no scrolling and no previous gesture detected*/
+        if(scroll_obj == NULL && proc->types.pointer.gesture_sent == 0) {
             if(proc->long_pr_sent == 0) {
```
Comment (kisvegabor) [inline src/core/lv_indev.c:976]: The condition should be added one line below to process the `else` part regardless to the gesture.
```suggestion
            if(proc->long_pr_sent == 0  && proc->types.pointer.gesture_sent == 0) {
```

**Review by kisvegabor — COMMENTED** [review]
Thank you, I was also about to add a similar fix.

Just a minor thing.

File: src/core/lv_indev.c (hunk 1 above)
Comment (somebodyLi) [reply src/core/lv_indev.c:976]: only this line is not enough? the click should be trig also ?

**Comment by kisvegabor** [comment]
Ah, you are right. I was lost in the `if`s :slightly_smiling_face: 

Thank you!

**Comment by kisvegabor** [comment]
I was about pressing the merge button when something came to my mind. Let say there is a chart (or any other object) which is not scrollable. You want an event on CLICK but as the user clicks by accident he sliders his finger as if it was a gesture. As the chart is not interested in the gestures the user would expect to get CLICK event even if the finger was slid. 

In general the user should be able to freely slide his finger and get CLICK in the end if he is no interested in the gestures. 

To solve it we could change `LV_OBJ_FLAG_GESTURE_BUBBLE` to `LV_OBJ_FLAG_GESTURE_ENABLE`. It would work almost like `LV_OBJ_FLAG_GESTURE_BUBBLE` (if it's enabled on an object it absorbs the gesture, if not enabled it propagates it to the parent). However, it can be disabled by default on all widgets (including screens). And finally we should send a gesture only if it was enabled on on object or one of its parent. This way we could know that whether the gesture was really processed/used and whether we should send CLICK or not.

What do you think?

**Comment by somebodyLi** [comment]
> get CLICK in the end if he is no interested in the gestures.

Sorry, I can't agree with you on this view .  `GESTURE` and `CLICK` anyway should be mutually exclusive

**Comment by kisvegabor** [comment]
The problem is that we don't know if the gesture was consumed or not. So we need a solution to indicate whether the user is interested in the gesture event or not. 

That's why I've suggested  to have `LV_OBJ_FLAG_GESTURE_ENABLE`. And once it's set, we will know that the gesture is consumed.

However (it just cam to my mind), what if the user is interested only in the LEFT gesture but the gestures of the other directions should ignored and treated as CLICK/SCROLL/etc.

**Pull request closed without merging**